lib/access-context.js

```diff
@@ -143,19 +144,9 @@ AccessContext.prototype.getUserId = function() {
     // or permit to resolve a valid user model
     var userModel = this.registry.findModel(p.type);
     if (!userModel) continue;
-    if (userModel.modelName === 'User') {
+    if (userModel.prototype instanceof BaseUser) {
       return p.id;
     }
```

Comment:
The way i get `Registry.getModel()`: `this.registry.getModel('User')` would actually return **one among** several models inheriting from built-in User model, not the built-in User Model itself

then i think `userModel.prototype instanceof BaseUser` would only return `true` for models being instances or inheriting from that particular User model

do you see things different?